### PR TITLE
Flatten model configuration (i.e. extend from GenerationCommonConfig

### PR DESCRIPTION
**Pull request by MichaelDoyle**
_Before you submit a pull request, please make sure you have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md)._

**This pull request is related to:**

- [X] A bug
- [ ] A new feature
- [ ] Documentation
- [X] Other (please specify)

**I have checked the following:**

- [X] I have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md);
- [X] I have added new tests (for bug fixes/features);
- [X] I have added/updated the documentation (for bug fixes / features).

**Description:**
Some time ago we, we moved to away from `custom` model parameters in favor of a flattened structure. This change will rectify this for all relevant generative models by creating custom configs schemas that extend from `GenerationCommonConfigSchema`. In the process, I added some better type safety which yielded quite a few bugs. 😄 

**Related issues:**
See also: https://github.com/firebase/genkit/commit/6ec91ee98691d2741ab7cf2cea45bbf7231ef2d9

### CODE REVIEW

**Review by Dabolus — DISMISSED** [review]
Awesome! Thanks for the huge contribution and for fixing the bugs along the way. Looks good to me 🚀

**Comment by MichaelDoyle** [comment]
Resolved some merge conflicts. Will need a second review. Thanks!

**Review by Dabolus — APPROVED** [review]

**Pull request merged**
